sycl/test-e2e/ESIMD/unified_memory_api/Inputs/atomic_update_slm.hpp

```diff
@@ -0,0 +1,720 @@
+    if constexpr (std::is_same_v<Op<uint64_t, N>, ImplCmpxchg<uint64_t, N>>) {
+      passed &= run_test<UseAcc, uint64_t, N, Op, UseMask>(q);
+    }
+  }
+  return passed;
+}
+
+template <int N, template <class, int> class Op, bool UseMask,
+          bool UsePVCFeatures, bool UseAcc>
+bool test_fp_types(queue q) {
+  bool passed = true;
+  if constexpr (UsePVCFeatures) {
+    if constexpr (std::is_same_v<Op<sycl::half, N>,
+                                 ImplLSCFmax<sycl::half, N>> ||
+                  std::is_same_v<Op<sycl::half, N>,
+                                 ImplLSCFmin<sycl::half, N>> ||
+                  std::is_same_v<Op<sycl::half, N>,
+                                 ImplLSCFcmpwr<sycl::half, N>>) {
+      auto dev = q.get_device();
+      if (dev.has(sycl::aspect::fp16)) {
+        passed &= run_test<UseAcc, sycl::half, N, Op, UseMask>(q);
+      }
+    }
+  }
+  passed &= run_test<UseAcc, float, N, Op, UseMask>(q);
+  return passed;
+}
+
+template <template <class, int> class Op, bool UseMask, bool UsePVCFeatures,
+          bool UseAcc, int SignMask = (Signed | Unsigned)>
+bool test_int_types_and_sizes(queue q) {
+  bool passed = true;
+  passed &= test_int_types<1, Op, UseMask, UsePVCFeatures, UseAcc, SignMask>(q);
+  passed &= test_int_types<2, Op, UseMask, UsePVCFeatures, UseAcc, SignMask>(q);
+  passed &= test_int_types<4, Op, UseMask, UsePVCFeatures, UseAcc, SignMask>(q);
+  passed &= test_int_types<8, Op, UseMask, UsePVCFeatures, UseAcc, SignMask>(q);
+  passed &=
+      test_int_types<16, Op, UseMask, UsePVCFeatures, UseAcc, SignMask>(q);
+
+  // Supported by LSC atomic:
+  if constexpr (UsePVCFeatures) {
+    passed &=
+        test_int_types<32, Op, UseMask, UsePVCFeatures, UseAcc, SignMask>(q);
+    passed &=
+        test_int_types<64, Op, UseMask, UsePVCFeatures, UseAcc, SignMask>(q);
+    // non power of two values are supported only in newer driver.
+    // TODO: Enable this when the new driver reaches test infrastructure
+    // (v27556).
+#if 0
+    passed &= test_int_types<12, Op, UseMask, UsePVCFeatures, UseAcc, SignMask>(q);
+    passed &= test_int_types<33, Op, UseMask, UsePVCFeatures, UseAcc, SignMask>(q);
+#endif
+  }
+
+  return passed;
+}
+
+template <template <class, int> class Op, bool UseMask, bool UsePVCFeatures,
+          bool UseAcc>
+bool test_fp_types_and_sizes(queue q) {
+  bool passed = true;
+  passed &= test_fp_types<1, Op, UseMask, UsePVCFeatures, UseAcc>(q);
+  passed &= test_fp_types<2, Op, UseMask, UsePVCFeatures, UseAcc>(q);
+  passed &= test_fp_types<4, Op, UseMask, UsePVCFeatures, UseAcc>(q);
+  passed &= test_fp_types<8, Op, UseMask, UsePVCFeatures, UseAcc>(q);
+  passed &= test_fp_types<16, Op, UseMask, UsePVCFeatures, UseAcc>(q);
+
+  // Supported by LSC atomic:
+  if constexpr (UsePVCFeatures) {
+    passed &= test_fp_types<32, Op, UseMask, UsePVCFeatures, UseAcc>(q);
+    passed &= test_fp_types<64, Op, UseMask, UsePVCFeatures, UseAcc>(q);
+#if 0
```

Comment:
Please copy that  TODO comment to this place too.

Reply:
Done.